File: mobile/ios/packages/town-crier-domain/Sources/Protocols/ZonePreferencesRepository.swift
/// Fetches and updates per-zone notification preferences.
///
/// Maps to `GET/PUT /v1/me/watch-zones/{zoneId}/preferences`.
/// The `updatePreferences` call is entitlement-gated on the API side --
/// Personal+ is required for status change and decision update toggles.
public protocol ZonePreferencesRepository: Sendable {
  func fetchPreferences(zoneId: String) async throws -> ZoneNotificationPreferences
  func updatePreferences(_ preferences: ZoneNotificationPreferences) async throws
}

File: mobile/android/domain/src/main/kotlin/uk/towncrierapp/domain/watchzones/ZonePreferencesRepository.kt
package uk.towncrierapp.domain.watchzones

/**
 * Fetches and updates per-zone notification preferences. Maps to
 * `GET/PUT /v1/me/watch-zones/{zoneId}/preferences`. Port of iOS
 * `ZonePreferencesRepository`.
 */
public interface ZonePreferencesRepository {
    public suspend fun fetchPreferences(zoneId: WatchZoneId): ZoneNotificationPreferences

    public suspend fun updatePreferences(preferences: ZoneNotificationPreferences)
}
